### ISSUE TITLE
Provide weight for loss

### ISSUE THREAD

**Issue by edizhuang** [open-mmlab/mmsegmentation#1685]
Hi,

How to provide/pass weights (on pixels of the image) to the loss (typically it is cross-entropy loss)?

And what is the right way to assign different weights to different classes?

Thanks,

**Comment by xiexinch** [comment]
Hi @edizhuang
You may add weight at the loss dict in a model config.
https://github.com/open-mmlab/mmsegmentation/blob/36984d98baeeea94831b3f21cf5065bd049a4d00/configs/_base_/models/fcn_r50-d8.py#L28

To assign different weights to different classes, you may add a parameter to the loss config.
https://github.com/open-mmlab/mmsegmentation/blob/36984d98baeeea94831b3f21cf5065bd049a4d00/mmseg/models/losses/cross_entropy_loss.py#L207
For example:
```python
loss_decode=dict(
            type='CrossEntropyLoss',
            use_sigmoid=False,
            loss_weight=1.0,
            class_weights=[0.1, 0.2, ......]) # weights for each class
```

**Comment by xiexinch** [comment]
Might see it at decode_head.py
https://github.com/open-mmlab/mmsegmentation/blob/14f6ec5314f4faf2970d4e6c30efda5998d382c9/mmseg/models/decode_heads/decode_head.py#L204

**Comment by edizhuang** [comment]
Hi,

I see pixel sampler does this weight on segmentation label (2D). 

If I want to specify OHEMPixelSampler, or customize my own sampler, which config should I update?

    PIXEL_SAMPLERS = Registry('pixel sampler')
    
    def build_pixel_sampler(cfg, **default_args):
        """Build pixel sampler for segmentation map."""
        return build_from_cfg(cfg, PIXEL_SAMPLERS, default_args)

Thanks,

**Comment by jiachen0212** [comment]
> Hi @edizhuang You may add weight at the loss dict in a model config.
> 
> https://github.com/open-mmlab/mmsegmentation/blob/36984d98baeeea94831b3f21cf5065bd049a4d00/configs/_base_/models/fcn_r50-d8.py#L28
> 
> To assign different weights to different classes, you may add a parameter to the loss config.
> 
> https://github.com/open-mmlab/mmsegmentation/blob/36984d98baeeea94831b3f21cf5065bd049a4d00/mmseg/models/losses/cross_entropy_loss.py#L207
> 
> 
> For example:
> ```python
> loss_decode=dict(
>             type='CrossEntropyLoss',
>             use_sigmoid=False,
>             loss_weight=1.0,
>             class_weight=[0.1, 0.2, ......]) # weights for each class
> ```

len(class_weight) == back_ground+len(class_list) ?
